chips/imxrt10xx/src/dma.rs

```diff
@@ -0,0 +1,784 @@
+        interfaces::{ReadWriteable, Readable, Writeable},
+        ReadOnly, ReadWrite, WriteOnly,
+    },
+    StaticRef,
+};
+
+use core::cell::Cell;
+use core::mem;
+use core::ops::Index;
+
+use crate::ccm;
+
+/// DMA Multiplexer.
+///
+/// The multiplexer is used for routing between DMA channels and hardware
+/// peripherals. It's a detail of `DmaChannel`.
+#[repr(C)]
+struct DmaMultiplexerRegisters {
+    /// Channel configuration registers, one per channel.
+    chcfg: [ReadWrite<u32, ChannelConfiguration::Register>; 32],
+}
+
+const DMA_MUX_BASE: StaticRef<DmaMultiplexerRegisters> =
+    unsafe { StaticRef::new(0x400E_C000 as *const DmaMultiplexerRegisters) };
+
+registers::register_bitfields![u32,
+    /// Each of the DMA channels can be independently enabled/disabled and associated
+    /// with one of the DMA slots (peripheral slots or always-on slots) in the system.
+    ///
+    /// Note: Setting multiple CHCFG registers with the same source value will result in
+    /// unpredictable behavior. This is true, even if a channel is disabled (ENBL==0).
+    ///
+    /// Note: Before changing the trigger or source settings, a DMA channel must be
+    /// disabled via CHCFGn[ENBL].
+    ChannelConfiguration [
+        /// Enables the channel for DMA Mux. The DMA has separate channel
+        /// enables/disables, which should be used to disable or reconfigure
+        /// a DMA channel.
+        ENBL OFFSET(31) NUMBITS(1) [],
+        /// Enables the periodic trigger capability for the triggered DMA channel.
+        ///
+        /// 0b - Triggering is disabled. If triggering is disabled and ENBL is set,
+        /// the DMA Channel will simply route the specified source to the DMA channel.
+        /// (Normal mode)
+        ///
+        /// 1b - Triggering is enabled. If triggering is enabled and ENBL is set,
+        /// the DMA_CH_MUX is in Periodic Trigger mode.
+        TRIG OFFSET(30) NUMBITS(1) [],
+        /// DMA Channel Always Enable
+        ///
+        /// Enables the DMA Channel to be always ON.
+        /// If TRIG bit is set, the module will assert request on every trigger.
+        ///
+        /// 0b - DMA Channel Always ON function is disabled
+        /// 1b - DMA Channel Always ON function is enabled
+        A_ON OFFSET(29) NUMBITS(1) [],
+        /// DMA Channel Source (Slot Number)
+        ///
+        /// Specifies which DMA source, if any, is routed to a particular DMA channel.
+        /// See the "DMA MUX Mapping" table in the "Interrupts, DMA Events, and XBAR
+        /// Assignments" chapter for details about DMA source and channel information.
+        SOURCE OFFSET(0) NUMBITS(7) []
+    ]
+];
+
+#[repr(C, align(32))]
+struct TransferControlDescriptor {
+    saddr: ReadWrite<u32>,
+    soff: ReadWrite<u16>, // Signed number
+    attr: ReadWrite<u16, TransferAttributes::Register>,
+    nbytes: ReadWrite<u32>, // Assumes minor loop mapping is disabled (EMLM = 0)
+    slast: ReadWrite<u32>,  // Signed number
```

Comment:
These comments are a bit confusing -- is there a reason we can't use `<i32>` here?

Reply:
cc @lschuermann -- shouldn't this be pretty easy to add now, with the new traits? Is there any reason not to do this?

Reply:
It's probably easy to add, though we must be careful on how we interact with bitfields then. We can't implement it outside of the library as either the trait (`IntLike`) or the type needs to be in the local crate for that to work. We might want to create a supertrait of `IntLike` which then explicitly enables bitfield-style manipulations on unsigned integers only, because semantically that might not make a ton of sense with signed integers.

Whether or not this is useful in general I don't know. Not being able to use bitfields for that takes away a major chunk of Tock-register's utility. I'm not strictly opposed, but casting seems easy enough if it's just rare cases.

Reply:
@mciantyre short term, I think leaving as `u16` with comments is fine. Long term, we should probably consider this part of the registers interface, but that's orthogonal to this PR.